controllers/runner_pod_controller.go

```diff
@@ -108,6 +117,24 @@ func (r *RunnerPodReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 	} else {
 		log.V(2).Info("Seen deletion-timestamp is already set")
 
+		if finalizers, removed := removeFinalizer(runnerPod.ObjectMeta.Finalizers, runnerLinkedResourcesFinalizerName); removed {
+			if err := r.cleanupRunnerLinkedPods(ctx, &runnerPod, log); err != nil {
+				log.Error(err, "cleanup failed")
+				return ctrl.Result{Requeue: true, RequeueAfter: 30 * time.Second}, nil
```

Comment:
We retry here, which I'm open to changing. The goal is to ensure that we do eventually clean up all of these resources, but that may come at a cost of getting "stuck" if for some reason we are unable to clean it up. I would love guidance on the suggested pattern here!